### PR TITLE
Cache results from slow python script test_labels.py

### PR DESCRIPTION
**Pull request by quantumsteve**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes

Describe what this PR changes and why.  If it closes an issue, link to it here
with [a supported keyword](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

Noticed that the python script test_labels.py is being rerun each time cmake is run. This caches the result in a unique variable

This is part of #2227. I hope we can further speed up generation of TEST_LABELS.

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Refactoring (no functional changes, no api changes)
- Build related changes

### Does this introduce a breaking change?

- No

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

File: CMake/test_labels.cmake [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
-    OUTPUT_VARIABLE TEST_LABELS_TEMP
-    RESULT_VARIABLE SUCCESS)
-  #MESSAGE("  Label script return value: ${SUCCESS}")
-  if(NOT ${SUCCESS} STREQUAL "0")
-    message("Warning: test labeling failed.  Test labeling error output:\n${TEST_LABELS_TEMP}")
-    set(TEST_LABELS_TEMP "")
-    #ELSE()
-    #  MESSAGE("  Test: ${TEST_NAME}")
-    #  MESSAGE("    ${TEST_LABELS_TEMP}")
+  if (DEFINED ${TEST_NAME}_${QMC_CUDA}_${QMC_COMPLEX}_${QMC_MIXED_PRECISION})
```
Comment (ye-luo) [inline CMake/test_labels.cmake:3]: `NOT DEFINED`?
Also please start with `TEST_LABELS_`
So `TEST_LABELS_${TEST_NAME}_${QMC_CUDA}_${QMC_COMPLEX}_${QMC_MIXED_PRECISION}`

**Review by ye-luo — CHANGES_REQUESTED** [review]

**Comment by ye-luo** [comment]
It will be more ugly if I do squash on github. Could you rebase and squash this PR?

**Comment by quantumsteve** [comment]
> It will be more ugly if I do squash on github. Could you rebase and squash this PR?

done

**Review by ye-luo — APPROVED** [review]
Great. Second cmake only took 6 sec now on my workstation.

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**